File: include/correlations/saturated-oil-density.rb

# coding: utf-8

# verificada con python!
begin_correlation("PobBradley", "SaturatedOilDensity", "Lb_ft3")
add_title("CALCULATION OF SATURATED OIL DENSITY")
add_parameter("yg",  "Sgg", "Gas specific gravity")
add_parameter("rs",  "SCF_STB", "Solution GOR")
add_parameter("bob",  "RB_STB", "Saturated oil formation volume factor")
add_parameter("yo",  "Sg_do", "Oil specific gravity")
add_author("Standard Equation")
add_ref("beggs:1987")
end_correlation()

################################################################        

File: include/correlations/saturated-oil-density.cc
Quantity<Lb_ft3> PobBradley__correlation__fct(const Quantity<Sgg>& yg,
                                              const Quantity<SCF_STB>& rs,
                                              const Quantity<RB_STB>& bob,
                                              const Quantity<Sg_do>& yo) {
  return PobBradley::get_instance()(yg, rs, bob, yo);
}

double PobBradley__correlation__valid__fct(const double& yg, const double& rs,
                                           const double& bob,
                                           const double& yo) {
  return PobBradley::get_instance()(yg, rs, bob, yo).raw();
}
